### PR TITLE
Implement storage and handlers for permalink feature.

### PR DESCRIPTION
**Pull request by dkumiszhan**
Left a couple of TODOs related to handling of custom types by sqlx library.
Will still need to figure out how to store int slices (`connkey`) and untyped field `decoded`.

### CODE REVIEW

File: backend/routes.go [hunk 1]
```diff
@@ -43,5 +43,10 @@ func newRouter(cs *server) (*httprouter.Router, error) {
 	r.PATCH("/queries/:id", handlers.CorsMiddleware(bearerMiddleware(cs.updateQueryHandler, os.Args[3])))
 	r.DELETE("/queries/:id", handlers.CorsMiddleware(bearerMiddleware(cs.deleteQueryHandler, os.Args[3])))
 
+	// event
+	r.POST("/events", handlers.CorsMiddleware(bearerMiddleware(cs.createEventHandler, os.Args[3])))
+	r.DELETE("/events/:id", handlers.CorsMiddleware(bearerMiddleware(cs.deleteEventHandler, os.Args[3])))
+	r.GET("/events", handlers.CorsMiddleware(bearerMiddleware(cs.getEventsHandler, os.Args[3])))
```
Comment (glaslos) [inline backend/routes.go:49]: We probably want to also have a route to get a specific event by ID

File: backend/entities/event.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	DstPort   int            `json:"dstPort,omitempty"`   // the connection destination port
+	Rule      string         `json:"rule,omitempty"`      // the rule that matched the connection
+	Handler   string         `json:"handler,omitempty"`   // the processing handler
+	Transport string         `json:"transport,omitempty"` // the transport used
+	Scanner   string         `json:"scanner,omitempty"`   // name of the scanner if detected
+	SensorID  string         `json:"sensorID,omitempty"`  // the id of the sensor
+	SrcHost   string         `json:"srcHost,omitempty"`   // the source IP address
+	SrcPort   string         `json:"srcPort,omitempty"`   // the source port
+	Timestamp string         `json:"timestamp,omitempty"` // the UTC timestamp of the connection
+	Decoded   types.JSONText `json:"decoded,omitempty"`   // a decoded version of the payload if available
```
Comment (glaslos) [inline backend/entities/event.go:21]: `types.JSONText` allows you to store and read the decoded object

File: backend/entities/event.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+import (
+	"github.com/jmoiron/sqlx/types"
+)
+
+type Event struct {
+	ID        string         `json:"id,omitempty"`
+	OwnerID   string         `json:"ownerID,omitempty"`
+	Payload   string         `json:"payload,omitempty"`
+	ConnKey   []int          `json:"connKey,omitempty"`   // identifier based on IP and source port
```
Comment (glaslos) [inline backend/entities/event.go:11]: We can omit this member

File: backend/routes.go (hunk 1 above)
Comment (dkumiszhan) [reply backend/routes.go:49]: Added a new route. It is not used from the frontend yet. Can work on that in a separate CL.

Comment (dkumiszhan) [reply backend/routes.go:49]: Added a new route but it is not being used by the frontend yet. Can work on that in a separate PR.

**Review by glaslos — APPROVED** [review]
LGTM

**Pull request merged**